### PR TITLE
Implement broadcast function

### LINKED ISSUE #511: Make broadcast function
We need a broadcast function which is called *explicitly*.
For example, shape of `x` is `()` and shape of `y` is (3, 4), user can call multiplication like this:
```
bx, by = F.broadcast(x, y)
xy = bx * by
```

Of course implicit broadcasting which is implemented in numpy and cupy is useful, but it is another problem. We need to discuss about implicit broadcasting in issue #228.

### PR DESCRIPTION
**Pull request by unnonouno**
fix #511 
Please merge #510 before this PR.

### CODE REVIEW

File: chainer/functions/array/broadcast.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    axis = tuple(i for i, sx in enumerate(x.shape) if sx == 1)
+    if len(axis) > 0:
+        return g.sum(keepdims=True, axis=axis)
+    else:
+        return g
+
+
+class Broadcast(function.Function):
+
+    """Function that broadcast given arrays."""
```
Comment (delta2323) [inline chainer/functions/array/broadcast.py]: broadcast**s**

File: chainer/functions/array/broadcast.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+def _backward_one(x, g):
+    if g is None:
+        xp = cuda.get_array_module(x)
+        return xp.zeros_like(x)
+
+    if g.ndim != x.ndim:
+        g = g.sum(axis=tuple(range(g.ndim - x.ndim)))
+        # An input variable is always an array, not a scalar.
+        # We need to convert a scalar value to a zero-dim array.
```
Comment (delta2323) [inline chainer/functions/array/broadcast.py]: Can we use `keepdims` to keep `g` `ndarray`?

**Comment by delta2323** [comment]
Could you add the test cases where the # of input array is more than 2?

File: tests/chainer_tests/functions_tests/array_tests/test_broadcast.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+class TestBroadcastScalar(TestBroadcast):
+
+    x_shape = (3, 2, 5)
+    y_shape = ()
+    out_shape = (3, 2, 5)
+
+
+class TestBroadcastTypeError(unittest.TestCase):
+
+    def test_invalid_shape(self):
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/array_tests/test_broadcast.py]: How about make two test fixtures two parametrized unit tests?

File: tests/chainer_tests/functions_tests/array_tests/test_broadcast.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    y_shape = (5,)
+    out_shape = (3, 2, 5)
+
+
+class TestBroadcastScalar(TestBroadcast):
+
+    x_shape = (3, 2, 5)
+    y_shape = ()
+    out_shape = (3, 2, 5)
+
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/array_tests/test_broadcast.py]: How about testing such cases?
* `x_shape` = (3, 2, 5), `y_shape` = (3, 2, 5)

**Comment by unnonouno** [comment]
You are right. I'll add tests for other sizes.
And, I think `size == 0` is illegal. I'll fix type check.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**